### PR TITLE
[react] Types for Popover API

### PR DESCRIPTION
**Pull request by eps1lon**
Types for https://github.com/facebook/react/issues/27479
Closes https://github.com/DefinitelyTyped/DefinitelyTyped/discussions/66550

### CODE REVIEW

**Comment by acusti** [comment]
i upgraded to the latest react/react-dom RC (`19.0.0-rc-4c2e457c7c-20240522`), but i'm getting typescript errors when trying to use any Popover API attributes, and i'm unable to import and use the new `ToggleEvent` type. looking at the releases for @types/react, i noticed that the latest @types/react v18.3.3 was released just before this was merged. is there a way to use canary or nightly types, especially for changes that have been released in the latest RC for a package?

**Comment by acusti** [comment]
just updating to answer my own question. i looked at the diff to see if i could extend the type defs to include these changes and noticed the [`react/canary.d.ts` lib def](https://github.com/eps1lon/DefinitelyTyped/blob/master/types/react/canary.d.ts) amongst the files changed. that peaked my curiosity, so i took a look at the file and found some very helpful instructions in the comment preamble.

tl;dr for others working in a typescript react app wanting to use the new Popover API: add `/// <reference types="react/canary" />` anywhere in your project (the main entry point made sense to me) and you will get all of the new Popover API types (at least if you have `@types/react` v18.3.3 installed).

File: types/react/canary.d.ts [hunk 1]
```diff
@@ -119,6 +121,21 @@ declare module "." {
         onTransitionStartCapture?: TransitionEventHandler<T> | undefined;
     }
 
+    type ToggleEventHandler<T = Element> = EventHandler<ToggleEvent<T>>;
+
+    interface HTMLAttributes<T> {
+        popover?: "" | "auto" | "manual" | undefined;
```
Comment (francislavoie) [inline types/react/canary.d.ts:127]: The type here is missing `"hint"` as per https://developer.mozilla.org/en-US/docs/Web/API/Popover_API/Using#creating_the_tooltips_with_popoverhint
